hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HoodieRealtimeInputFormatUtils.java

```diff
@@ -87,6 +93,20 @@
           fsCache.put(metaClient, fsView);
         }
         HoodieTableFileSystemView fsView = fsCache.get(metaClient);
+        // fetch virtual key params if required.
```

Comment:
@vinothchandar : quick question. here we have a N no of metaclients, one per partition path (line 86). 
Is it safe to assume all partitions have same schema? Then, should I move this outside of foreach and just pick first entry and find the tableConfig to populate hoodieVirtualKeyInfo. If not, we are fetching the schema for every partition to find the hoodieVirtualKeyInfo

Reply:
I think we can just do it once.